WalletWasabi.Fluent/Helpers/TransactionFeeHelper.cs

```diff
@@ -117,27 +117,57 @@ public static TimeSpan CalculateConfirmationTime(double targetBlock)
 		return time;
 	}
 
-	public static bool TryGetMaximumPossibleFeeRate(decimal percentageOfOverpayment, Wallet wallet, FeeRate currentFeeRate, out FeeRate maximumPossibleFeeRate)
+	/// <summary>
+	/// Seeks for the maximum possible fee rate that the transaction can pay.
+	/// </summary>
+	/// <remarks>The method does not throw any exception.</remarks>
+	/// <remarks>Stores the found fee rate in the received <see cref="TransactionInfo"/> object. </remarks>
+	/// <returns>True if the seeking was successful, False if not.</returns>
+	public static async Task<bool> TrySetMaxFeeRateAsync(Wallet wallet, TransactionInfo info)
 	{
-		maximumPossibleFeeRate = FeeRate.Zero;
+		var maxFeeRate =
+			await Task.Run(() =>
+			{
+				var found = FeeHelpers.TryGetMaxFeeRate(wallet, info.Destination, info.Amount, info.Recipient, info.FeeRate, info.Coins, info.SubtractFee, out var maxFeeRate, false);
+
+				return found ? maxFeeRate! : new FeeRate(0m);
+			});
 
-		if (percentageOfOverpayment <= 0 || !TryGetFeeEstimates(wallet, out var feeEstimates))
+		if (EnsureFeeRateIsPossible(wallet, maxFeeRate))
 		{
-			return false;
+			info.MaximumPossibleFeeRate = maxFeeRate;
+			info.FeeRate = maxFeeRate;
+			info.ConfirmationTimeSpan = TryEstimateConfirmationTime(wallet, maxFeeRate, out var estimate)
+				? estimate.Value
+				: TimeSpan.Zero;
+
+			return true;
 		}
 
-		var maxPossibleFeeRateInSatoshiPerByte = (currentFeeRate.SatoshiPerByte / percentageOfOverpayment) * 100;
-		maximumPossibleFeeRate = new FeeRate(maxPossibleFeeRateInSatoshiPerByte);
+		return false;
+	}
+
+	/// <summary>
+	/// Temporary solution for making sure if a fee rate can be found in the fee chart.
+	/// It is needed otherwise we cannot predict the confirmation time and the fee chart would crash.
+	/// TODO: Remove this hack when the issues mentioned above are fixed.
+	/// </summary>
```

Comment:
This is currently in master too, will fix it in a separate PR.

Reply:
Instead of having `TrySetMaxFeeRateAsync` to potentially return false and fail, why here instead of returning false we don't return the maximum value possible?

Reply:
The thing is it wouldn't fail, the transaction could be easily built and broadcast. The problem is on our side.
Let's say in our estimation the last item is 5 sat/vbyte for a week confirmation.

If the max fee rate for the transaction is <5, the fee slider dialog would crash as every value that is higher than the max fee rate would be removed from the chart, which would result in an empty chart.
The second problem is it is not really handled what to say as a confirmation time on the transaction preview for this case.
Fixing these would be the goal for the next PR. It wouldn't be a good idea the modify the algorithm to return an incorrect value for the reasons above.